osgar/drivers/lora.py

```diff
@@ -35,6 +35,17 @@ def parse_lora_packet(packet):
        with the actual payload data. In the case of missing transmitted
        history header None is returned.
     """
+    if b'|' not in packet or len(packet) < 2:  # 3rd party or debug packet?
+        return None, packet
+
+    # restrict ID to only five supported devices
+    if packet[0] not in [ord(str(x)) for x in range(1, 6)]:
+        return None, packet
+
+    # make sure, that '|' is the second character
+    if packet[1] != ord('|'):
+        return None, packet
+
     s = packet.split(b'|')
     try:
         return [int(x) for x in s[:-1]], s[-1].strip()
```

Comment:
Why don't we validate just this data? We can check that all address fields are within the expected set instead of checking just the first one etc. We also print error to stdout and only on ValueError and not on any others.

Reply:
OK, I will add refactoring here. Regarding print of "errors" - it is hard - there are "tons" of some debug messages which we really do not want to print, so I will probably also hide the ValueError prints.

Reply:
Fixed.

Reply:
Yeah, I just meant we should strive to be consistent. What is the motivation for validating only the first address and not all of them?